services/src/main/java/org/keycloak/protocol/oid4vc/issuance/TimeProvider.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.protocol.oid4vc.issuance;
+
+/**
+ * Interface to provide the current time
+ */
+public interface TimeProvider {
```

Comment:
Maybe it is possible to avoid this class and instead directly use Keycloak `Time` utility?

It allows to call `setTimeOffset` in the tests inside integration arquillian testsuite (See `AbstractKeycloakTest.setTimeOffset` for more details). 

On the other hand, I can see that you use the `StaticTimeProvider` in the test. So in that case if "static" time is needed and time offset is not sufficient, it should be fine to have the `TimeProvider` like you have. But can be nice to avoid that though...

Reply:
Yeah, so I see TimeProvider can be good aproach in that case. Thanks for the clarification! Resolving this point.